### PR TITLE
Add DeleteRelease method

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

This is needed for chart-operator e2e. Also added some refactoring in the local e2e test to prepare ground for the upcoming pull tarball test.

### CODE REVIEW

File: apprclient.go [hunk 1]
```diff
@@ -176,6 +178,28 @@ func (c *Client) PromoteChart(name, release, channel string) error {
 	return nil
 }
 
+// DeleteRelease removes a release from the server.
+func (c *Client) DeleteRelease(name, release string) error {
```
Comment (xh3b4sd) [inline apprclient.go]: Would be cool to order the methods alphabetically.

**Review by xh3b4sd — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]

File: spec.go [hunk 2]
```diff
@@ -3,8 +3,10 @@ package apprclient
 const (
 	httpClientTimeout = 5
 
-	// status returned on a successful request to a CNR server.
+	// status returned on a successful GET request to a CNR server.
```
Comment (rossf7) [inline spec.go]: okStatus

File: apprclient.go (hunk 1 above)
Comment (fgimenez) [reply apprclient.go]: Sure, done.

File: spec.go (hunk 2 above)
Comment (fgimenez) [reply spec.go]: Good catch, I've changed both to include the name of the constants.

**Pull request merged**